tests/integration/azure_blob/test_azure_blob.py

```diff
@@ -246,18 +247,82 @@ def test_list_dir_rm(self, use_datastore_profile, auth_method):
         assert self.object_file.split("/")[-1] not in dir_dataitem.listdir()
         file_dataitem.delete()  # should not raise an error
 
-    @pytest.mark.parametrize("use_datastore_profile", (True, False))
-    def test_blob_upload(self, use_datastore_profile):
+    @pytest.mark.parametrize(
+        "auth_method ,use_datastore_profile", generated_pytest_parameters
+    )
+    def test_blob_upload(self, use_datastore_profile, auth_method):
+        # The upload is done by a different connector than fsspec, so it requires checking every authentication method.
         self.setup_before_test(
-            use_datastore_profile=use_datastore_profile,
-            auth_method="fsspec_conn_str" if use_datastore_profile else "env_conn_str",
+            use_datastore_profile=use_datastore_profile, auth_method=auth_method
         )
         upload_data_item = mlrun.run.get_dataitem(self.object_url, self.storage_options)
         upload_data_item.upload(self.test_file)
 
         response = upload_data_item.get()
         assert response.decode() == self.test_string
 
+    @pytest.mark.parametrize("use_datastore_profile", (True, False))
+    def test_large_upload(self, use_datastore_profile):
+        # Time-consuming test, so every authentication method is checked in test_blob_upload, which is faster.
```

Comment:
Maybe we can also skip the `use_datastore_profile` parameterization then? I find it hard to think of a scenario where that would matter specifically for large files.